src/Administration/Controller/AdministrationController.php

```diff
@@ -109,6 +112,19 @@ public function index(Request $request, Context $context): Response
         ]);
     }
 
+    #[Route(path: '/%shopware_administration.path_name%/sso/auth', name: 'administration.sso.auth', defaults: ['auth_required' => false], methods: ['GET'])]
+    public function onSsoAuth(Request $request): RedirectResponse
+    {
+        $key = $request->get('key');
+        $random = $request->get('rdm');
```

Comment:
Is this the endpoint for the SSO buttons? If so why we only store the random value in the session, that we receive from the button href? We can't trust data from the browser. Instead we should compare it here, because of CSRF:

https://en.wikipedia.org/wiki/Cross-site_request_forgery